### DISCUSSION TITLE
is NextUI SEO friendly? (for next.js 14)

### DISCUSSION THREAD

**Discussion by rabieeali** [heroui-inc/heroui#2653] [Q&A]
Hello there!
I am going to use NextUI for an e-commerce website with Next.js version 14 (app router), and according to these lines from the docs:

"Next.js 13 introduces a new app/ directory structure. By default it uses Server Components. As NextUI components use React hooks, we added the use client; at build time, so you can import them directly in your React Server Components (RSC)."

so if "use client" is set for every component by default, are NextUI components SEO friendly?
I will be glad if you could explain more on this topic!
thanks

**Comment by dvzrd** [comment]
Your components should have nothing to do with SEO as you'll be configuring your metadata inside your layout and page files.

Read this for more info:

- https://nextjs.org/docs/app/building-your-application/optimizing/metadata
- https://nextjs.org/docs/app/api-reference/functions/generate-metadata

**Reply by Hahlh** [reply]
Links that aren't crawlable are one of many examples where this isn't true.

**Comment by mkrystkowicz** [comment]
Depends, some SEO signals for google are checking site performance and stuff, when it comes to for example INP metric I wrote a topic on this one. - [HERE](https://github.com/nextui-org/nextui/discussions/2981)

Not sure how important that signal is when google ranks the page, but might turn out that you won't be the first one because of this metric.
